src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -0,0 +1,46 @@
+package org.folio.rest.impl;
+import io.vertx.core.Context;
+import me.escoffier.vertx.completablefuture.VertxCompletableFuture;
+import org.folio.rest.jaxrs.model.InvoiceLineCollection;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+import static org.folio.invoices.utils.HelperUtils.encodeQuery;
+import static org.folio.invoices.utils.HelperUtils.handleGetRequest;
+import static org.folio.invoices.utils.ResourcePathResolver.INVOICE_LINES;
+import static org.folio.invoices.utils.ResourcePathResolver.resourcesPath;
+
+public class InvoiceLineHelper extends AbstractHelper {
+
+  private static final String GET_INVOICE_LINES_BY_QUERY = resourcesPath(INVOICE_LINES) + "?limit=%s&offset=%s%s&lang=%s";
+
+  InvoiceLineHelper(Map<String, String> okapiHeaders, Context ctx, String lang) {
+    super(getHttpClient(okapiHeaders), okapiHeaders, ctx, lang);
+  }
+
+  public CompletableFuture<InvoiceLineCollection> getInvoiceLines(int limit, int offset, String query) {
+    CompletableFuture<InvoiceLineCollection> future = new VertxCompletableFuture<>(ctx);
+    try {
+      String queryParam = isEmpty(query) ? EMPTY : "&query=" + encodeQuery(query, logger);
+      String endpoint = String.format(GET_INVOICE_LINES_BY_QUERY, limit, offset, queryParam, lang);
+      handleGetRequest(endpoint, httpClient, ctx, okapiHeaders, logger)
+        .thenAccept(jsonInvoiceLines -> {
+          if (logger.isInfoEnabled()) {
+            logger.info("Successfully retrieved invoice lines: {}", jsonInvoiceLines.encodePrettily());
+          }
+          future.complete(jsonInvoiceLines.mapTo(InvoiceLineCollection.class));
```

Comment:
This operation potentially might become blocking in case a lot of lines found. Please consider to use `VertxCompletableFuture.supplyBlockingAsync`

Reply:
updated to use supplyBlockingAsync